wormhole-connect/src/hooks/useTransactionHistory.ts

```diff
@@ -0,0 +1,193 @@
+import { useCallback, useEffect, useState } from 'react';
+import { useSelector } from 'react-redux';
+
+import useTransactionHistoryWHScan from 'hooks/useTransactionHistoryWHScan';
+import useTransactionHistoryMayan from 'hooks/useTransactionHistoryMayan';
+
+import type { Transaction } from 'config/types';
+import type { RootState } from 'store';
+
+type Props = {
+  page?: number;
+  pageSize?: number;
+};
+
+const useTransactionHistory = (
+  props: Props,
+): {
+  transactions: Array<Transaction> | undefined;
+  error: Array<string>;
+  isFetching: boolean;
+  hasMore: boolean;
+} => {
+  const { page = 0, pageSize = 30 } = props;
+
+  // Keeping separate indexes to track the last rendered item in respective transaction sets
+  const [mayanIndex, setMayanIndex] = useState(0);
+  const [whScanIndex, setWHScanIndex] = useState(0);
+
+  // We need to keep the last fetched pages from each APIs
+  // as the number of items in a page from each API can be different when sorted by time.
+  const [mayanPage, setMayanPage] = useState(page);
+  const [whScanPage, setWHScanPage] = useState(page);
```

Comment:
Hey @artursapek , I totally agree and have thought about a separate adaptor to handle the page and indexes for each API fetch. If you think that'd not be a throw away code and can be moved into the SDK for the Transaction history implementation there, I already had this in mind. Otherwise I think it's better to keep all that implementation inside the hook where we can retire easily once SDK had the functionality. LMK 🙏